interfaces/azure_storage/v0/README.md

```diff
@@ -0,0 +1,65 @@
+# `azure_storage`
+
+## Usage
+
+This relation interface describes the expected behaviour of any charm claiming to be able to interact with Azure Storage protocol.
+This relation interface should be used for both Azure Blob Storage and Azure Data Lake Storage (Gen 2). This interface will be accomplished using the provider library, although charm developers are free to provide alternative libraries as long as they fulfil the behavioural and schematic requirements described in this document.
+
+## Direction
+
+```mermaid
+flowchart TD
+    Provider -- container, storage-account, secret-key, connection-protocol, endpoint, path --> Requirer
+```
+
+As with all Juju relations, the `azure` interface consists of two parties: a Provider (object storage charm) and a Requirer (application charm). The Provider will be expected to provide new unique credentials (along with `storage-account`, `container`, `connection-protocol` and other fields), which can be used to access the actual object storage.
+
+## Behaviour
+
+Both the Requirer and the Provider must adhere to criteria to be compatible with this interface.
+
+### Provider
+- It is expected to provide `storage-account`, `container` and `secret-key` fields corresponding to a storage account, container and the secret key corresponding to an Azure Storage account when a relation joins. As of now, the container is not automatically created when the relation is joined.
```

Comment:
instead of re-describing what should be clear and documented in the schema, focus in this section on the behaviours that the two roles should implement. e.g. `The provider is expected to generate a storage pool and share with the requirer all the required credentials and configurations as per the schema.`

Reply:
The provider here does nothing but re-echo its configuration. I can probably rephrase this into something better.